demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/SingleMatchController.scala

```diff
@@ -104,10 +113,20 @@ class SingleMatchController @Inject()(
         )
       ) map { resp: AddressBySearchResponseContainer =>
         val filledForm = SingleMatchController.form.fill(SingleSearchForm(addressText,formatText))
+
+        val classCodes = mutable.Map[String, String]()
+
+        resp.response.addresses.map{address =>
```

Comment:
You are just transforming a list into a map
If the list is empty, you will get an empty map
This way you avoid mutable state